container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/user_services_functions/start_user_services.go

```diff
@@ -552,6 +555,17 @@ func createStartServiceOperation(
 			envVars[key] = strings.Replace(envVars[key], privateIPAddrPlaceholder, privateIPAddrStr, unlimitedReplacements)
 		}
 
+		// TODO clean this hack up
+		// this path will only be hit if `files_to_be_moved` is set in Starlark; which is a hidden property
+		// used by compose transpilation
+		if len(filesToBeMoved) > 0 {
```

Comment:
The `files_to_be_moved` hack is to enable volumes that mount files instead of directories. Right now, files artifact expansion always mounts things as directories and not just files. As opposed to rearchitecting files artifact expansion, @h4ck3rk3y implemented `files_to_be_moved` which mounts the files to `/tmp` directory on the container and then updates the `entrypoint` and `command` of the container (in container engine lib) to `mv` those files to the proper file location.